draft-deepspace-ip-architecture.xml

```diff
@@ -216,6 +216,23 @@
     </section>
     <section>
       <name>Transport</name>
+      <t>Modern transport protocols and stacks share similar algorithms for common transport needs.  This section first addresses general transport issues, and then addresses use of specific individual transport protocols.  While there have been academic research experiments in developing new protocols specifically for interplanetary transport, they are not directly considered here because the goal is to leverage Internet standards and commonly available software.</t>
+      <section anchor="general-transport">
+        <name>General Transport Issues</name>
+        <t>Internet transport protocols and transport stacks have developed to meet the needs for different classes of applications on the terrestrial Internet, using similar algorithms with parameters that are tuned for typical conditions.  Some of these algorithms might simply be paramterized differently for interplanetary network paths, while other algorithms may have fundamental challenges.  In some cases, minor adjustments (or no adjustment) will suffice for Earth-lunar networking, but Earth-Mars and other cases pose more significant challenges.</t>
+        <ul>
+          <li><t>Protocol Negotiation / "Happy Eyeballs" - Due to presence of both IPv4 and IPv6 in the terrestrial Internet, applications or transport stacks may include "happy eyeballs" capabilities <xref target="RFC8305"/> that make use of multiple DNS queries and connection attempts for different addresses and address families returned.  Downsides to this in interplanetary applications might include (1) the additional load on interplanetary DNS, (2) unnecessary use of network capacity for paralel connection attempts (e.g. if 0-RTT data transmission is used), and (3) unnecessary additional transport state maintained for an extended period of time.  If applications for interplanetary use initially rely on either fixed IP addresses (instead of DNS names) or a single address family (e.g. IPv6), then there should be no downsides to presence of happy eyeballs algorithms within the transport stack.  If happy eyeballs is present and triggered, then the values in section 8 of <xref target="RFC8305"/> should be carefully considered and tuned based on the DNS and path expectations (e.g. resolution delay, first address family count, connection attmept delay, minimum connection attempt delay, and maximum connection attempt delay) as most of these values are not likely to be appropriate even for Earth-lunar paths.</t></li>
```

Comment:
This is assuming that DNS queries ever go interplanetary.  This is obviously something to be avoided to begin with.  Should we start with the DNS discussion?  I would assume that all space systems are firewalled from the rest of the Internet and so there is no real need for full public DNS.  For systems behind the firewall, we should have a full-blown set of terrestrial authoritative servers. For off-Earth systems, we should have secondary servers that are pre-loaded with all of their regular correspondents. Doesn't this resolve the happy eyeballs issue?

Reply:
Yes, these could be good early architecture assumptions that simplify things to remove potentially problematic things from happening (e.g. happy eyeballs).  However, I believe interconnection w/ Internet is inevitable, it just may either take a bit longer time, or happen accidentally.  So, I think we should state assumptions for this architecture in early/initial deployments of interplanetary IP, and note that it could need to evolve if those assumptions are violated in the future.